### PR TITLE
mark a match

### LINKED ISSUE #4: Highlight search term
Wouldn’t it be cool to highlight the searchterm within in the result?

Like this:
image

### PR DESCRIPTION
**Pull request by Haroenv**
This adds a `<mark>` around the matched words.

However I notice that it starts replacing html with its equivalent without tags for some reason, maybe someone else can shine a light on this.

cc @mischah, @aarohmankad and @lukyvj

fixes #4 

Possible issues
- removes event listeners and everything attached to the code

The problem is that if you check with `innerHTML`, that it 
1. also matches `&amp;` for example
2. matches inside tags, and can break them
3. requires you to set `innerHTML` which breaks all event listeners etc.
4. http://stackoverflow.com/a/1732454/3185307

That means you'd better search in `textContent` and replace `innerHTML` which fixes problems 2 and maybe 1, but definitely breaks the `listeners`. 

TODO:
- [x] try out regex
- [x] don't mess up the `html`
- [x] highlight the matched word
- [x] document the feature
  - [x] demo
  - [x] jsdoc
  - [x] readme

### CODE REVIEW

**Comment by Haroenv** [comment]
Finally got something working here. Will merge once I wrote about this feature.

**Comment by Haroenv** [comment]
Great, this can be merged after another look

File: index.html [hunk 1]
```diff
@@ -281,7 +281,7 @@
   </script>
 
   <!-- SVG icons -->
-  <svg viewBox="0 0 100 100"  xmlns="http://www.w3.org/2000/svg">
+  <svg viewBox="0 0 100 100"  xmlns="http://www.w3.org/2000/svg" display="none">
```
Comment (LukyVj) [inline index.html]: Oh shit, you can remove this. I forgot to remove it.

Comment (Haroenv) [reply index.html]: done

**Pull request merged**